ZeeguuAPI/ZeeguuAPI/ZeeguuAPI.swift

```diff
@@ -0,0 +1,334 @@
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				completion(langCode: response!)
+			} else {
+				completion(langCode: nil)
+			}
+		}
+	}
+	
+	/// Sets the language code of the native langugage of the logged in user.
+	///
+	/// - parameter newLanguageCode: The language code of the language that will be the new native language.
+	/// - parameter completion: A block that will receive a boolean indication if the request succeeded.
+	public func setNativeLanguage(newLanguageCode: String, completion: (success: Bool) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(success: false)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.NativeLanguage, pathComponents: [newLanguageCode], method: HTTPMethod.POST, parameters: nil)
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil && response! == "OK") {
+				completion(success: true)
+			} else {
+				completion(success: false)
+			}
+		}
+	}
+	
+	/// Returns the language code of the learned and native langugage of the logged in user.
+	///
+	/// - parameter completion: A block that will receive a `JSON` object, which contains the dictornary with language codes of the learned and native language.
+	public func learnedAndNativeLanguage(completion: (dict: JSON?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(dict: nil)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.LearnedAndNativeLanguage, pathComponents: nil, method: HTTPMethod.GET, parameters: nil)
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				completion(dict: JSON.parse(response!))
+			} else {
+				completion(dict: nil)
+			}
+		}
+	}
+	
+	/// Returns the language codes of all available languages that the Zeeguu API supports.
+	///
+	/// - parameter completion: A block that will receive a `JSON` object, which contains the array with the language codes.
+	public func availableLanguages(completion: (array: JSON?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(array: nil)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.AvailableLanguages, pathComponents: nil, method: HTTPMethod.GET, parameters: nil)
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				completion(array: JSON.parse(response!))
+			} else {
+				completion(array: nil)
+			}
+		}
+	}
+	
+	/// Returns the words that the user is currently studying.
+	///
+	/// - parameter completion: A block that will receive a `JSON` object, which contains the list of words.
+	public func studyingWords(completion: (array: JSON?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(array: nil)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.UserWords, pathComponents: nil, method: HTTPMethod.GET, parameters: nil)
```

Comment:
I have done Smalltalk, ObjectiveC, and C... and this convention of Swift is puzzling for me... Either one should be using keyword arguments, or not, but combining them like this... I don't know. I would find function names to be much more readable if we would use the keyword also for the first argument, and removing it from the method name. In this case something like 

     self.apiRequest(endpoint: ZeeguuAPIEndpoint.UserWords,
                               pathComponents: nil,
     ...

What do you think?